distutils/tests/test_unixccompiler.py

```diff
@@ -214,6 +215,38 @@ def gcvs(*args, _orig=sysconfig.get_config_vars):
             sysconfig.customize_compiler(self.cc)
         self.assertEqual(self.cc.linker_so[0], 'my_cc')
 
+    @unittest.skipIf(sys.platform == 'win32', "can't test on Windows")
+    def test_cc_overrides_ldshared_for_cxx_correctly(self):
+        # Issue #18080:
+        # ensure that setting CC env variable also changes default linker
+        def gcv(v):
+            if v == 'LDSHARED':
+                return 'gcc-4.2 -bundle -undefined dynamic_lookup '
+            elif v == 'CXX':
+                return 'g++-4.2'
+            return 'gcc-4.2'
+
+        def gcvs(*args, _orig=sysconfig.get_config_vars):
+            if args:
+                return list(map(sysconfig.get_config_var, args))
+            return _orig()
+
+        sysconfig.get_config_var = gcv
+        sysconfig.get_config_vars = gcvs
+        with patch.object(self.cc, 'spawn', return_value=None) as mock_spawn, \
+                patch.object(self.cc, '_need_link', return_value=True) as mock_need, \
+                patch.object(self.cc, 'mkpath', return_value=None) as mock_mkpath, \
+                EnvironmentVarGuard() as env:
+            env['CC'] = 'ccache my_cc'
+            env['CXX'] = 'my_cxx'
+            del env['LDSHARED']
+            sysconfig.customize_compiler(self.cc)
+            self.assertEqual(self.cc.linker_so[0:2], ['ccache','my_cc'])
+            self.cc.link(None, [], 'a.out', target_lang='c++')
+            call_args = mock_spawn.call_args[0][0]
+            if len(call_args) >= 2:
+                assert(call_args[:2] != ['my_cxx', 'my_cc'])
```

Comment:
The only assertion in this test is that the args are `!=` to some value. That leaves it open to a lot of potential alternative values. Can you clarify what this test is asserting? Is this test merely working to assert that "setting CC env variable also changes default linker" or is it attempting to do other things (e.g. ensure that multiple commands in the compiler are honored)?

Reply:
This was asserting that the bug reported didn't happen. I didn't add an assert that this would equal to the expected value because I wasn't sure what the expected value was for AIX.